### PR TITLE
Slug 209 rename experience date search to experience slot search

### PR DESCRIPTION
**Pull request by eballantine**
See easol[#PR10710](https://github.com/easolhq/easol/pull/10710)

We are updating the naming of experience 'dates' to experience 'slots' to future-proof for the slots being aware of time in addition to date.

In these docs I am only updating the naming. Where there is explanation given about the returned objects being dates, I'm leaving this as is, as it's still factually true.

I did some huge find and replaces, then divvied up into commits for ease of reviewing but it will probably be best to squash once approved as they aren't atomic.

### CODE REVIEW

File: docs/reference/objects/product/modifier.md [hunk 1]
```diff
@@ -190,7 +190,7 @@ number
 {: .label .fs-1 }
 
 The remaining stock for the modifier. If the modifier has unlimited inventory this will return `nil`.
-- When accessed through an [experience date]({% link docs/reference/objects/product/experience_date.md %}), it will return the remaining stock for the modifier on the given date.
+exp- When accessed through an [experience slot]({% link docs/reference/objects/product/experience_slot.md %}), it will return the remaining stock for the modifier on the given date.
```
Comment (CatalinaTudor) [inline docs/reference/objects/product/modifier.md:193]: 👀 This `exp` seems like it doesn't belong here.

File: docs/reference/objects/product/index.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ timestamp
 {: .label .fs-1 }
 
 - If the product has a fixed date this returns the start date of the event as a timestamp, this can then be used in conjunction with Liquid's [built-in filters](https://shopify.github.io/liquid/filters/date/).
-- If the product is an experience with multiple dates, this returns the start date of the next upcoming or ongoing date if there is one, or the most recent date if all dates are in the past.
+- If the product is an experience with multiple slots, this returns the start date of the next upcoming or ongoing date if there is one, or the most recent date if all slots are in the past.
```
Comment (CatalinaTudor) [inline docs/reference/objects/product/index.md:68]: **question** Should this say: `this returns the start date of the next upcoming or ongoing slot if there is one`? as `the start date of the next upcoming or ongoing date` sounds a bit weird 🤔

**Review by CatalinaTudor — APPROVED** [review]
Looks good ✨ Just left a couple of minor comments 👍

File: docs/reference/objects/product/modifier.md (hunk 1 above)
Comment (eballantine) [reply docs/reference/objects/product/modifier.md:193]: Catalina eagle eyes at work, thank you so much 🙏

File: docs/reference/objects/product/index.md (hunk 2 above)
Comment (eballantine) [reply docs/reference/objects/product/index.md]: Yes can definitely be more clear here, it now reads:
`If the product is an experience with multiple slots, this returns the start date of the next upcoming or ongoing slot if there is one, or the start date of the most recent slot if all slots are in the past.`
Verbose, but accurate.

Comment (CatalinaTudor) [reply docs/reference/objects/product/index.md]: I think that's much better, even if a bit longer 🚀

**Pull request merged**